content/blog/eng/automation-testing-a-step-back.md

```diff
@@ -0,0 +1,114 @@
+
+Why is it important to write automated tests? I asked myself and I did some research because the answer to this question was not obvious to me. I knew it was important but I didn't know why. So I decided to try to motivate it starting from a point of view as impartial as possible.
+
+After some research I came to the conclusion that the benefits of the tests are four: they can avoid regression, they can count as documentation, they can help write better code and they can help find bugs during development.
+
+Let's focus on one of them at the time and analyze it.
+
+## Avoid regression
+
+> When you implemented a feature in your app, you likely ran the code to verify that it worked as expected. You performed a test, albeit a manual test. As you continued to add and update features, you probably also continued to run your code and verify it works. But doing this manually every time is tiring, prone to mistakes, and does not scale.  
+Computers are great at scaling and automation! So developers at companies large and small write automated tests, which are tests that are run by software and do not require you to manually operate the app to verify the code works.
+
+This is the introduction of a [Google test codelab](https://codelabs.developers.google.com/codelabs/advanced-android-kotlin-training-testing-basics/#0). It not only defines in a pretty straightforward way what automated tests are, but it also tells why you should do that. The reason mentioned here is only the first of the four on the list that I have compiled before, but this is the main reason.
+
+So we write automated tests to save time in the future, it's an investment. We suppose that when we add more features the old ones need to be checked again, and we invest our time in order to save more in the future, but in the short term we will not have a return.
+
+If we imagine a big project with many developers it becomes more clear, we cannot check manually everything every time (if we want to release often or with simplicity). So if we want to release soon and we have no automation tests we end up testing only what seems most important at the moment, exposing ourselves to the risk of regression.
+
+## Count as documentation
+
+This is an easy point. Documentation can be important (less than working software, see agile manifesto) but it's a pain for most and takes a lot of time to do it well. But if tests are written in a certain way they can have the same information of product documentation, that evolves and adapts as the software grows. This way the software can be self-documented.
+
+## Help write better code
+
+This is a very generic sentence, what do I mean by "better code"? First say that software as every other built thing has a quality. A program can be better than another that does the same things based on some parameters (performance, usability and many others), we can divide these parameters into 2 big groups: the one that users notice and the one that users don't.
```

Comment:
First, remember that software quality, as that of other built things, can be measured. A program can be better than another one that does the same things based on some parameters (performance, usability and many others).  We can divide these parameters into 2 big groups: the one that users notice and the one that users don't.